src/mixins/canvas_events.mixin.js

```diff
@@ -207,40 +212,179 @@
       this.__onLongPress && this.__onLongPress(e, self);
     },
 
+    /**
+     * supports native like text dragging
+     * @private
+     * @param {DragEvent} e
+     */
+    _onDragStart: function (e) {
+      var activeObject = this.getActiveObject();
+      if (activeObject && typeof activeObject.onDragStart === 'function' && activeObject.onDragStart(e)) {
+        this._dragSource = activeObject;
+        var options = { e: e, target: activeObject };
+        this.fire('dragstart', options);
+        activeObject.fire('dragstart', options);
+        addListener(this.upperCanvasEl, 'drag', this._onDragProgress);
+        return;
+      }
+      e.preventDefault();
+      e.stopPropagation();
+    },
+
+    /**
+     * @private
+     */
+    _renderDragStartSelection: function () {
+      this._dragSource && typeof this._dragSource.renderDragStartSelection === 'function'
+        && this._dragSource.renderDragStartSelection();
+    },
+
+    /**
+     * supports native like text dragging
+     * https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/Drag_operations#finishing_a_drag
+     * @private
+     * @param {DragEvent} e
+     */
+    _onDragEnd: function (e) {
+      var didDrop = e.dataTransfer.dropEffect !== 'none',
+          dropTarget = didDrop ? this._activeObject : undefined,
+          options = {
+            e: e,
+            target: this._dragSource,
+            subTargets: this.targets,
+            dragSource: this._dragSource,
+            didDrop: didDrop,
+            dropTarget: dropTarget
+          };
+      removeListener(this.upperCanvasEl, 'drag', this._onDragProgress);
+      this.fire('dragend', options);
+      this._dragSource && this._dragSource.fire('dragend', options);
+      delete this._dragSource;
+      // we need to call mouse up synthetically because the browser won't
+      this._onMouseUp(e);
+    },
+
+    /**
+     * fire `drag` event on canvas and drag source
+     * @private
+     * @param {DragEvent} e
+     */
+    _onDragProgress: function (e) {
+      var options = {
+        e: e,
+        dragSource: this._dragSource,
+        dropTarget: this._draggedoverTarget
+      };
+      this.fire('drag', options);
+      this._dragSource && this._dragSource.fire('drag', options);
+    },
```

Comment:
while this fires like 30 times per second when you do not move the mouse, and like 120 times per second when moving the mouse. Just to say never put anything substantial.
I understand we need this to render the cursor, otherwise i would happily skip this.

Reply:
I added this to comply with the spec.
AFAIK it has no side effects. It's meant for the dev only, fabric doesn't use it.
The cursor is handled by the browser's drag event functionality.

Reply:
I meant the fabricJS text cursor that indicates the drop position of text. It seems to me that without this event we can't draw it, and so we need it.

Reply:
> The cursor is handled by the browser's drag event **native** functionality.

Reply:
We do not need it

Reply:
Did you use in particular? can you make a good use case? otherwise let's not add it.